### DISCUSSION TITLE
404 There isn't a GitHub Pages site here

### DISCUSSION THREAD

**Discussion by yuvrajsharma97** [community/community#22809] [Other Feature Feedback, Questions, & Ideas]
<h1>404</h1>
<p><strong>There isn’t a GitHub Pages site here.</strong></p>
<p>I am getting this error form an already published website when I changed the repository to private and again to public. Please Help.</p>

**Comment by meetnandu05** [comment]
<p>You will have to check the enforce HTTPS option available in the manage access tab of your repo settings.</p>

**Reply by havanlinh99** [reply]
There isn't a GitHub Pages site here.

If you're trying to publish one, read the full documentation to learn how to set up GitHub Pages for your repository, organization, or user account.

GitHub Status — @githubstatus

**Comment by otter-computer** [comment, marked as answer]
<p>Hi <a class="mention" href="/imyuvi1234">@imyuvi1234</a>, welcome to the GitHub Support Community!</p>
<p>GitHub Pages sites aren’t able to be hosted from private repositories when you’re on the free plan. When you change your repositories visibility settings any Pages sites hosted from this repository will be disabled. To get your site working again you’ll need to make sure your repository is set to public again, head to the repository settings page, make sure that GitHub Pages is enabled, then push a fresh commit to your repository. Doing this will trigger a rebuild of your site and should make your site available after it’s finished rebuilding.</p>

**Reply by surya382** [reply]
i made the repository public again also made a fresh commit to master branch but still error 404 persisits

**Reply by mcknuggie** [reply]
I also have this same error

**Reply by mcknuggie** [reply]
I fixed the issue by re-running all jobs under github actions

**Reply by goutamthakur** [reply]
I was facing the same issue since my plan was downgraded from Pro to Free, I made my repo public, added a new commit, and my page was live again.

**Reply by farhanahmedtech** [reply]
check the URL

**Reply by Sudhir22sep** [reply]
It is still showing the 404 page after making it public and commiting, the is happening with the vercel.app

**Comment by gowlemn** [comment]
Sometimes when you fail to pay and your plan is switch to free plan, there are some errors even after you renew your subscription. If you want your site to be published without any further commits this here is what worked for me. Simply make your repository public and then private again. Hope this helps. Thank you.
-gowlemn-

**Comment by NYP-FYP-DEV** [comment]
I am not able to host https://nyp-fyp-dev.github.io/ReactPolicyMgmt even though I had already published successfully and also in the Github Pages setting it didn't display the homepage of my url. Please help. Thank you.

**Reply by pradip292** [reply]
same problem

**Reply by grzlx69** [reply]
> та же проблема

**Comment by YsdCastro24** [comment]
I have the same issue, except that my repository is public, not private. The main page, which is the index, loads fine, but when I select another section, I get this error. However, when I compile the code locally in the workspace, everything works correctly.

**Reply by kc9304** [reply]
any solution for it?

**Reply by SbuYandisa** [reply]
Any solutions for this, I am getting the same error.

**Reply by v-meesala** [reply]
I faced the same issue. 
Try renaming the repo to something like "usrname.github.io"
Then again deploy from local by using "npm run deploy" cmd
wait for couple of minutes for github to reflect that in the website.
if still didn't work, try running the deploy cmd from the directory where your section content is present.

**Reply by zzq8** [reply]
> I faced the same issue. Try renaming the repo to something like "usrname.github.io" Then again deploy from local by using "npm run deploy" cmd wait for couple of minutes for github to reflect that in the website. if still didn't work, try running the deploy cmd from the directory where your section content is present.

https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#types-of-github-pages-sites

**Reply by Kenssou** [reply]
please any solution to this yet ?

**Comment by MuhammadHamzaAslam** [comment]
response #1

**Comment by MuhammadHamzaAslam** [comment]
response #1

**Comment by kamal321854** [comment]
![Uploading 1753173072344-01.jpeg…]()

**Comment by rydzn27** [comment]
Same problem. I have a repository named portfolio, and it used to publish correctly at:

https://rydzn27.github.io/portfolio/

The website was working fine a few days ago, but now it only shows:

404
There isn't a GitHub Pages site here.

I tried unpublishing it and publishing it again to no avail.

I checked my other pages, and they also show error 404:
https://rydzn27.github.io/mir-h/
https://rydzn27.github.io/landing-page/

Even though GitHub Pages shows the site as “published,” accessing the URL still returns a 404.

Does anyone know what might be causing this or how to fix it?

**Reply by arnxv0** [reply]
These links work for me. I was also facing the same issue on https://arnavdewan.dev , and I suspect it has to do with the TTL of the records; it should resolve itself within 24 hours.

**Comment by jashwanth1128** [comment]
This happens because GitHub Pages does not serve sites from private repositories unless Pages is explicitly re-enabled.
When you changed the repository from public → private → public, GitHub disabled Pages automatically, which breaks the published URL and results in:

404 – There isn’t a GitHub Pages site here

Your site files are still there, but Pages is no longer active.

How to fix it (step by step):

Open your GitHub repository

Go to Settings → Pages

Under Build and deployment:

Set Source to:

Deploy from a branch

Select:

Branch: main (or master)

Folder: /root (or /docs, whichever you used earlier)

Click Save

GitHub will redeploy the site automatically.

Important checks (don’t skip):

Ensure index.html exists in the selected folder

Wait 1–2 minutes after saving (deployment is not instant)

Hard refresh the page (Ctrl + F5)

Verify the URL format:

https://<username>.github.io/<repository-name>/


If it still shows 404:

Check the Actions tab to confirm Pages deployment succeeded

If you use a custom domain:

Re-add the domain in Pages settings

Ensure the CNAME file still exists

Try disabling Pages once, save, then re-enable it again

Why this happens:

Switching a repo to private disables GitHub Pages

Switching back to public does NOT automatically restore Pages

Manual reconfiguration is required

**Reply by c0tton-fluff** [reply]
I did all that you mentioned ... still shows 404 ... something broke half way through the day after it worked like a charm and I was able to see it online. Then ... no clue ... I changed DNS settings, I reset it all, I rerun all checks ... everything ...

**Reply by jashwanth1128** [reply]
Bro, this is not a GitHub Pages enable issue anymore.
Since the site already worked earlier, Pages itself is fine.
The real problem is broken domain verification after DNS changes.
That’s why GitHub is showing:
404 – There isn’t a GitHub Pages site here
What actually fixes it (this works):
1) Remove the custom domain completely
Go to Repo → Settings → Pages
Delete the custom domain
Save and wait 2–3 minutes
Open the default URL:
https://<username>.github.io/<repo-name>/
If this works, Pages is healthy.
2) Fix DNS properly (most important part)
If using apex domain (example.com)
Keep ONLY these A records:
Copy code

185.199.108.153
185.199.109.153
185.199.110.153
185.199.111.153
 Remove all old GitHub IPs
Remove parked-domain IPs
Remove ALL AAAA (IPv6) records
If using www.example.com
Copy code

CNAME → <username>.github.io
Do not mix A + CNAME on the same host
3) Re-add the custom domain
Go back to Settings → Pages
Add the domain again
Save
Wait 5–10 minutes (DNS is slow)
Enable Enforce HTTPS only after it appears
4) Ensure CNAME file exists In repo root, file named CNAME with one line only:
Copy code

example.com
(or www.example.com) No spaces, no https.
5) Clear cache
Check in incognito / mobile data
DNS changes take time, don’t keep refreshing immediately
Why earlier steps didn’t work
Changing repo visibility disables Pages,
but changing DNS after Pages works breaks domain verification silently.
GitHub does not warn clearly — it just returns 404.
This exact sequence fixes it.
I’ve seen this happen many times.

**Reply by c0tton-fluff** [reply]
Thank you very much for your reply. So testing the 1st point, https://.github.io does not work and gives out the 404, same error ... I gave it over 12 hours just in case but it seems to be the Pages problem now ... Unsure how to proceed, might give it a day or so to relax :)

Edit: I ended up rebuilding it from scratch. I used Actions instead of gh-pages, managed without an issue ... maybe it was my issue that something got messed up completely. Now, page is live and all is well. Thanks for the help before!
